### PR TITLE
Restore me delete api

### PR DESCRIPTION
**Pull request by lucychen-grafana**
THIS FOR HACKATHON ONLY
**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/manage-dashboards/state/actions.ts [hunk 1]
```diff
@@ -359,10 +359,14 @@ export function getFolderById(id: number): Promise<{ id: number; title: string }
   return getBackendSrv().get(`/api/folders/id/${id}`);
 }
 
-export function deleteDashboard(uid: string, showSuccessAlert: boolean) {
+export function deleteDashboard(uid: string, showSuccessAlert: boolean): Promise<{ id: number; title: string }> {
```
Comment (evictorero) [inline public/app/features/manage-dashboards/state/actions.ts:362]: Why did you add the promise response?

File: public/app/features/manage-dashboards/state/actions.ts [hunk 2]
```diff
@@ -359,10 +359,14 @@ export function getFolderById(id: number): Promise<{ id: number; title: string }
...
   return getBackendSrv().delete<DeleteDashboardResponse>(`/api/dashboards/uid/${uid}`, { showSuccessAlert });
 }
 
+export function deleteDashboardPermanently(uid: string, showSuccessAlert: boolean) {
```
Comment (evictorero) [inline public/app/features/manage-dashboards/state/actions.ts:366]: This should be part of another PR.

File: public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardModal.tsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ const DeleteDashboardModalUnconnected = ({ hideModal, cleanUpDashboardAndVariabl
 
   const [, onConfirm] = useAsyncFn(async () => {
     await deleteDashboard(dashboard.uid, true);
-    cleanUpDashboardAndVariables();
+    // cleanUpDashboardAndVariables();
```
Comment (evictorero) [inline public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardModal.tsx:32]: Should this line be commented?

File: pkg/api/api.go [hunk 4]
```diff
@@ -454,7 +454,9 @@ func (hs *HTTPServer) registerRoutes() {
 		// Dashboard
 		apiRoute.Group("/dashboards", func(dashboardRoute routing.RouteRegister) {
 			dashboardRoute.Get("/uid/:uid", authorize(ac.EvalPermission(dashboards.ActionDashboardsRead)), routing.Wrap(hs.GetDashboard))
-			dashboardRoute.Delete("/uid/:uid", authorize(ac.EvalPermission(dashboards.ActionDashboardsDelete)), routing.Wrap(hs.DeleteDashboardByUID))
+			// dashboardRoute.Delete("/uid/:uid", authorize(ac.EvalPermission(dashboards.ActionDashboardsDelete)), routing.Wrap(hs.DeleteDashboardByUID))//TODO: new endpoint for hard delete
+			dashboardRoute.Delete("/uid/:uid", authorize(ac.EvalPermission(dashboards.ActionDashboardsDelete)), routing.Wrap(hs.SoftDeleteDashboard))
+			dashboardRoute.Post("/uid/:uid", authorize(ac.EvalPermission(dashboards.ActionDashboardsWrite)), routing.Wrap(hs.RestoreDashboard))
```
Comment (evictorero) [inline pkg/api/api.go:459]: What do you think of changing the restore endpoint to something like: 
`PATCH: /dashboards/uid/:uid/trash`  and use something similar for the hard delete: `DELETE:  /dashboards/uid/:uid/trash`

File: pkg/api/dashboard.go [hunk 5]
```diff
@@ -272,6 +272,70 @@ func (hs *HTTPServer) getDashboardHelper(ctx context.Context, orgID int64, id in
...
+	dash, rsp := hs.getDashboardHelper(c.Req.Context(), c.SignedInUser.GetOrgID(), 0, uid)
+	if rsp != nil {
+		return rsp
+	}
+
+	return response.JSON(http.StatusOK, util.DynMap{
+		"title":   dash.Title,
+		"message": fmt.Sprintf("Dashboard %s is restored", dash.Title),
+		"id":      dash.ID,
+		"deleted": dash.Deleted,
```
Comment (evictorero) [inline pkg/api/dashboard.go:306]: This value should be always empty in the restore.

Comment (lucychen-grafana) [reply pkg/api/dashboard.go:306]: Ah yes, I was going to remove that line, i was just using it to verify in postman

File: public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardModal.tsx (hunk 3 above)
Comment (lucychen-grafana) [reply public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardModal.tsx:32]: This method does [a lot of actions](https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/state/actions.ts#L32-L48) to clean up the dashboard including removing panels and variables and such. If it runs, recovering it would be difficult

Comment (evictorero) [reply public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardModal.tsx:32]: I would expect all of those deletions to be related to the frontend. The entire business logic should be in the backend endpoint. @juanicabanas can you help us confirming what is the purpose of that cleanup method?

Comment (juanicabanas) [reply public/app/features/dashboard/components/DeleteDashboard/DeleteDashboardModal.tsx:32]: let me take a look

File: public/app/features/manage-dashboards/state/actions.ts (hunk 1 above)
Comment (lucychen-grafana) [reply public/app/features/manage-dashboards/state/actions.ts]: Copy and paste accident; i have deleted it
